Document remote interaction properly (#946)
iterative/datachain: docs/references/remotes.md

>        if ``token='anon'``, the value is ignored.
>
>    - access : one of {'read_only', 'read_write', 'full_control'}
>        Full control implies read/write as well as modifying metadata,
>        e.g., access control.
>
>    - token: None, dict or string
>        (see description of authentication methods, from link above)
>
>    - consistency: 'none', 'size', 'md5'
>        Check method when writing files. Can be overridden in open().
>
>    - cache_timeout: float, seconds
>        Cache expiration time in seconds for object metadata cache.
>        Set cache_timeout <= 0 for no caching, None for no cache expiration.
>
>    - secure_serialize: bool (deprecated)
>
>    - requester_pays : bool, or str default False
>        Whether to use requester-pays requests. This will include your
>        project ID `project` in requests as the `userProject`, and you'll be
>        billed for accessing data from requester-pays buckets. Optionally,
>        pass a project-id here as a string to use that as the `userProject`.
>
>    - session_kwargs: dict
>        passed on to ``aiohttp.ClientSession``; can contain, for example,
>        proxy settings.
>

> >    - endpoint_url: str
> >        If given, use this URL (format protocol://host:port , *without* any
> >        path part) for communication. If not given, defaults to the value
> >        of environment variable "STORAGE_EMULATOR_HOST"; if that is not set
> >        either, will use the standard Google endpoint.
> >

Reviewer: ```suggestion
- `endpoint_url`: `string` (default: `None`)

    If given, use this URL (format: `protocol://host:port`, *without* any
    path part) for communication. If not given, defaults to the value
    of environment variable `"STORAGE_EMULATOR_HOST"`; if that is not set
    either, will use the standard Google endpoint.

```

> >    - default_location: str
> >        Default location where buckets are created, like 'US' or 'EUROPE-WEST3'.
> >        You can find a list of all available locations here:
> >        https://cloud.google.com/storage/docs/locations#available-locations
> >

Reviewer: ```suggestion
- `default_location`: `str` (default: `None`)

    Default location where buckets are created, like `"US"` or `"EUROPE-WEST3"`.
    You can find a list of all available locations here:
    https://cloud.google.com/storage/docs/locations#available-locations

```

> >    - version_aware: bool
> >        Whether to support object versioning. If enabled this will require the
> >        user to have the necessary permissions for dealing with versioned objects.
> 

Reviewer: ```suggestion
- `version_aware`: `bool` (default: `False`)

    Whether to support object versioning. If enabled this will require the
    user to have the necessary permissions for dealing with versioned objects.

```

### Azure Blob Storage

DataChain uses [adlfs](https://fsspec.github.io/adlfs/) to interact with Azure Blob Storage. Authentication can be achieved by using any of the method described at [adlfs documentation](https://github.com/fsspec/adlfs?tab=readme-ov-file#setting-credentials). You can also pass the following configuration parameters to the adlfs filesystem as client_config dictionary.

> 
> >	- account_name: str
> >        The storage account name. This is used to authenticate requests
> >        signed with an account key and to construct the storage endpoint. It
> >        is required unless a connection string is given, or if a custom
> >        domain is used with anonymous authentication.
> >

Reviewer: ```suggestion
- `account_name`: `str` (default: `None`)

    The storage account name. This is used to authenticate requests
    signed with an account key and to construct the storage endpoint. It
    is required unless a connection string is given, or if a custom
    domain is used with anonymous authentication.

```

> >    - account_key: str
> >        The storage account key. This is used for shared key authentication.
> >        If any of account key, sas token or client_id is specified, anonymous access
> >        will be used.
> >

Reviewer: ```suggestion
- `account_key`: `str` (default: `None`)

    The storage account key. This is used for shared key authentication.
    If any of account key, sas token or client_id is specified, anonymous access
    will be used.

```

> >    - sas_token: str
> >        A shared access signature token to use to authenticate requests
> >        instead of the account key. If account key and sas token are both
> >        specified, account key will be used to sign. If any of account key, sas token
> >        or client_id are specified, anonymous access will be used.
> >

Reviewer: ```suggestion
- `sas_token`: `str` (default: `None`)

    A shared access signature token to use to authenticate requests
    instead of the account key. If account key and sas token are both
    specified, account key will be used to sign. If any of account key, sas token
    or client_id are specified, anonymous access will be used.

```

> >    - request_session: Session
> >        The session object to use for http requests.
> >

Reviewer: ```suggestion
- `request_session`: `requests.Session` (default: `None`)

    The session object to use for http requests.

```

> >    - connection_string: str
> >        If specified, this will override all other parameters besides
> >        request session. See
> >        http://azure.microsoft.com/en-us/documentation/articles/storage-configure-connection-string/
> >        for the connection string format.
> >

Reviewer: ```suggestion
- `connection_string`: `str` (default: `None`)

    If specified, this will override all other parameters besides
    request session. See
    http://azure.microsoft.com/en-us/documentation/articles/storage-configure-connection-string/
    for the connection string format.

```

> >    - credential: azure.core.credentials_async.AsyncTokenCredential or SAS token
> >        The credentials with which to authenticate.  Optional if the account URL already has a SAS token.
> >        Can include an instance of TokenCredential class from azure.identity.aio.
> >

Reviewer: ```suggestion
- `credential`: `azure.core.credentials_async.AsyncTokenCredential` or SAS token (default: `None`)

    The credentials with which to authenticate. Optional if the account URL already has a SAS token.
    Can include an instance of TokenCredential class from azure.identity.aio.

```

> >    - blocksize: int
> >        The block size to use for download/upload operations. Defaults to hardcoded value of
> >        ``BlockBlobService.MAX_BLOCK_SIZE``
> >

Reviewer: ```suggestion
- `blocksize`: `int` (default: `None`)

    The block size to use for download/upload operations. Defaults to hardcoded value of
    `BlockBlobService.MAX_BLOCK_SIZE`

```

> >    - client_id: str
> >        Client ID to use when authenticating using an AD Service Principal client/secret.
> >

Reviewer: ```suggestion
- `client_id`: `str` (default: `None`)

    Client ID to use when authenticating using an AD Service Principal client/secret.

```

> >    - client_secret: str
> >        Client secret to use when authenticating using an AD Service Principal client/secret.
> >

Reviewer: ```suggestion
- `client_secret`: `str` (default: `None`)

    Client secret to use when authenticating using an AD Service Principal client/secret.

```

> >    - tenant_id: str
> >        Tenant ID to use when authenticating using an AD Service Principal client/secret.
> >

Reviewer: ```suggestion
- `tenant_id`: `str` (default: `None`)

    Tenant ID to use when authenticating using an AD Service Principal client/secret.

```

> >    - anon: boolean, optional
> >        The value to use for whether to attempt anonymous access if no other credential is
> >        passed. By default (``None``), the ``AZURE_STORAGE_ANON`` environment variable is
> >        checked. False values (``false``, ``0``, ``f``) will resolve to `False` and
> >        anonymous access will not be attempted. Otherwise the value for ``anon`` resolves
> >        to ``True``.
> >

Reviewer: ```suggestion
- `anon`: `boolean` (default: `None`)

    The value to use for whether to attempt anonymous access if no other credential is
    passed. By default (`None`), the `AZURE_STORAGE_ANON` environment variable is
    checked. False values (`false`, `0`, `f`) will resolve to `False` and
    anonymous access will not be attempted. Otherwise the value for `anon` resolves
    to `True`.

```

> >    - default_fill_cache: bool = True
> >        Whether to use cache filling with open by default
> >

Reviewer: ```suggestion
- `default_fill_cache`: `bool` (default: `True`)

    Whether to use cache filling with open by default

```

> >    - default_cache_type: string ('bytes')
> >        If given, the default cache_type value used for "open()".  Set to none if no caching
> >        is desired.  Docs in fsspec
> >

Reviewer: ```suggestion
- `default_cache_type`: `string` (default: `"bytes"`)

    If given, the default cache_type value used for `open()`. Set to `None` if no caching
    is desired. Docs in fsspec.

```

> >    - version_aware : bool (False)
> >        Whether to support blob versioning.  If enable this will require the
> >        user to have the necessary permissions for dealing with versioned blobs.
> >

Reviewer: ```suggestion
- `version_aware`: `bool` (default: `False`)

    Whether to support blob versioning. If enable this will require the user to have the
    necessary permissions for dealing with versioned blobs.

```

> >    - assume_container_exists: Optional[bool] (None)
> >        Set this to true to not check for existence of containers at all, assuming they exist.
> >        None (default) means to warn in case of a failure when checking for existence of a container
> >        False throws if retrieving container properties fails, which might happen if your
> >        authentication is only valid at the storage container level, and not the
> >        storage account level.
> >

Reviewer: ```suggestion
- `assume_container_exists`: `bool` (default: `None`)

    Set this to `True` to not check for existence of containers at all, assuming they exist.
    `None` (default) means to warn in case of a failure when checking for existence of a container.
    `False` throws if retrieving container properties fails, which might happen if your
    authentication is only valid at the storage container level, and not the
    storage account level.

```

> >    - max_concurrency:
> >        The number of concurrent connections to use when uploading or downloading a blob.
> >        If None it will be inferred from fsspec.asyn._get_batch_size().
> >

Reviewer: ```suggestion
- `max_concurrency`: `int` (default: `None`)

    The number of concurrent connections to use when uploading or downloading a blob.
    If `None` it will be inferred from `fsspec.asyn._get_batch_size()`.

```

> >    - timeout: int
> >        Sets the server-side timeout when uploading or downloading a blob.
> >

Reviewer: ```suggestion
- `timeout`: `int` (default: `None`)

    Sets the server-side timeout when uploading or downloading a blob.

```

> >    - connection_timeout: int
> >        The number of seconds the client will wait to establish a connection to the server
> >        when uploading or downloading a blob.
> >

Reviewer: ```suggestion
- `connection_timeout`: `int` (default: `None`)

    The number of seconds the client will wait to establish a connection to the server
    when uploading or downloading a blob.

```

> >    - read_timeout: int
> >        The number of seconds the client will wait, between consecutive read operations,
> >        for a response from the server while uploading or downloading a blob.
> >

Reviewer: ```suggestion
- `read_timeout`: `int` (default: `None`)

    The number of seconds the client will wait, between consecutive read operations,
    for a response from the server while uploading or downloading a blob.

```

> >    - account_host: str
> >        The storage account host. This string is the entire url to the for the storage
> >        after the https://, i.e. "https://{account_host}". This parameter is only
> >        required for Azure clouds where account urls do not end with "blob.core.windows.net".
> >        Note that the account_name parameter is still required.

Reviewer: ```suggestion
- `account_host`: `str` (default: `None`)

    The storage account host. This string is the entire url to the for the storage
    after the `https://`, i.e. `"https://{account_host}"`. This parameter is only
    required for Azure clouds where account urls do not end with `"blob.core.windows.net"`.
    Note that the `account_name` parameter is still required.
```

> ### Huggingface

Reviewer: ```suggestion
### Huggingface

```

> DataChain uses [huggingface_hub](https://pypi.org/project/huggingface-hub/) to interact with huggingfaace. You can pass the following parameters to client config to interact with huggingface.

Reviewer: ```suggestion
DataChain uses [huggingface_hub](https://pypi.org/project/huggingface-hub/) to interact with Hugging Face. You can pass the following parameters to client config to interact with Hugging Face.
```

> >	- token (`str` or `bool`, *optional*):
> >            A valid user access token (string). Defaults to the locally saved
> >            token, which is the recommended method for authentication (see
> >            https://huggingface.co/docs/huggingface_hub/quick-start#authentication).
> >            To disable authentication, pass `False`.
> >

Reviewer: ```suggestion
- `token`: `str` or `bool` (default: `None`)

    A valid user access token (string). Defaults to the locally saved
    token, which is the recommended method for authentication (see
    https://huggingface.co/docs/huggingface_hub/quick-start#authentication).
    To disable authentication, pass `False`.

```

> >    - endpoint (`str`, *optional*):
> >            Endpoint of the Hub. Defaults to <https://huggingface.co>.

Reviewer: ```suggestion
- `endpoint`: `str` (default: `None`)
    
    Endpoint of the Hub. Defaults to `https://huggingface.co`.
```
